servicetalk-concurrent-api/docs/modules/ROOT/pages/blocking-implementation.adoc

```diff
@@ -20,15 +20,17 @@ Everything inside ServiceTalk is somehow connected to one of the three asynchron
 and `Completable`. Since these sources are the building blocks for program control flow if they provide safety
 guarantees for blocking code execution these guarantees apply outside the scope of preventing blocking code from
 executing on event loop thread. This approach is designed to make the task of ensuring we don't block the event loop
-threads less error prone, and also allows for certain optimizations around thread context propagation and re-use.
+threads less error-prone, and also allows for certain optimizations around thread context propagation and re-use.
 
 == Threads and asynchronous sources
 
-An asynchronous source has to decide at two points about which thread will be used:
+An asynchronous source has two important decisions to make about thread usage:
 
-1. Thread which is used to do the actual work related to a source. eg: for an HTTP client, the work is to send an HTTP
+1. Which thread or executor will be used to do the actual work related to a source. eg: for an HTTP client, the work is to send an HTTP
 request and read the HTTP response.
-2. Thread which is used to interact with the `Subscriber` corresponding to its `Subscription`s.
+2. Which thread or executor will be used to interact with the `Subscriber` corresponding to its `Subscription`s.
+
+In sometimes a single specific thread will be used but in most cases a thread from an `Executor` pool will be used.
```

Comment:
`In sometimes` -> `Sometimes` or `On some occasions` (English is my second language, but feels the sentence is not right here).

Reply:
You are correct, it is bad English.